sonic_platform_base/sfp_base.py

```diff
@@ -0,0 +1,204 @@
+#
+# sfp_base.py
+#
+# Abstract base class for implementing a platform-specific class with which
+# to interact with a SFP module in SONiC
+#
+
+import sys
+from . import device_base
+
+
+class SfpBase(device_base.DeviceBase):
+    """
+    Abstract base class for interfacing with a SFP module
+    """
+
+    def get_transceiver_info(self):
+        """
+        Retrieves transceiver info of this SFP
+
+        Returns:
+            A dict which contains following keys/values :
+        ========================================================================
+        keys                       |Value Format   |Information	
+        ---------------------------|---------------|----------------------------
+        type                       |1*255VCHAR     |type of SFP
+        hardwarerev                |1*255VCHAR     |hardware version of SFP
+        serialnum                  |1*255VCHAR     |serial number of the SFP
+        manufacturename            |1*255VCHAR     |SFP venndor name
+        modelname                  |1*255VCHAR     |SFP model name
+        Connector                  |1*255VCHAR     |connector information
+        encoding                   |1*255VCHAR     |encoding information
+        ext_identifier             |1*255VCHAR     |extend identifier
+        ext_rateselect_compliance  |1*255VCHAR     |extended rateSelect compliance
+        cable_length               |INT            |cable length in m
+        mominal_bit_rate           |INT            |nominal bit rate by 100Mbs
+        specification_compliance   |1*255VCHAR     |specification compliance
+        vendor_date                |1*255VCHAR     |vendor date
+        vendor_oui                 |1*255VCHAR     |vendor OUI
+        ========================================================================
+        """
+        return NotImplementedError
+
+    def get_transceiver_bulk_status(self):
```

Comment:
didn't see the fields that are supported now which are fetched from DOM, like voltage, temperature, rx_power, tx_bias, etc.

Reply:
I have added temperature, voltage, TX bias, TX power and RX power into sfp_base.py to support DOM status.  They have also been included as part of the bulk_status.   Please review.  Thanks